lib/services/ngsi/ngsiService.js

```diff
@@ -426,11 +426,20 @@ function castJsonNativeAttributes(payload) {
 function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, callback) {
```

Comment:
I wonder if some other request appart from sending updates to CB could potencially have the same problem described in #733. Maybe looking for `/v2/entities` in the code could provide some hints.

What do you think @dcalvoalonso?

Reply:
Hi @fgalan, it is possible... I will review this.

Reply:
```
$ grep -R '/v2/entities' lib/
lib/services/ngsi/ngsiService.js:        '/v2/entities/' + entityName + '/attrs',
lib/services/ngsi/ngsiService.js:    var url = '/v2/entities/' + entityName + '/attrs';
lib/services/devices/deviceService.js:        url: config.getConfig().contextBroker.url + '/v2/entities?options=upsert',
lib/services/devices/deviceService.js:        options.url = deviceData.cbHost + '/v2/entities?options=upsert';
lib/services/devices/deviceService.js:        options.url = 'http://' + deviceData.cbHost + '/v2/entities?options=upsert';
lib/services/devices/deviceService.js:        url: config.getConfig().contextBroker.url + '/v2/entities/' + String(deviceData.name) + '/attrs',
lib/services/devices/deviceService.js:        options.url = deviceData.cbHost + '/v2/entities/' + String(deviceData.name) + '/attrs';
lib/services/devices/deviceService.js:        options.url = 'http://' + deviceData.cbHost + '/v2/entities/' + String(deviceData.name) + '/attrs';
````
Now I'll have a closer look to each one...

Reply:
In ngsiService.js the first one correspond to sendUpdateValueNgsi2() already included in this PR. The second one corresponds to sendQueryValueNgsi2() and I think that `?type` should be also added there.

In deviceService.js the three first occurences are in createInitialEntityNgsi2() so they don't need `?type` (entity type is already included in the request payload in this case). The three last ocurrences are in updateEntityNgsi2() and probably that one could also have ambiguity problems so `?type` should be included.

@dcalvoalonso what do you think? Do you agree with my analysis or is something wrong?

Reply:
Great. So let's wait to @danvilmot to finish the work with tests (and maybe this: https://github.com/telefonicaid/iotagent-node-lib/pull/734#discussion_r249471301) then you can continue from that part once this gets merged. Thanks!

NTC